genConfig.py

```diff
@@ -0,0 +1,146 @@
+import json
+from pprint import pprint
+import math
+from copy import deepcopy
+from commonFunctions import make_sure_path_exists
+
+def scanCombinations(scanParams):
+    myParam = scanParams.keys()[0]
+
+    r = dict(scanParams)
+    del r[myParam]
+
+    list=[]
+    myList=[]
+
+    if len(r) is not 0:
+        list=scanCombinations(r)
+
+    min = scanParams[myParam][0]
+    max = scanParams[myParam][1]
+    if max < min:
+        tmpVal = max
+        max = min
+        min = tmpVal
+    entries = 2
+    if len(scanParams[myParam]) == 3:
+        entries = scanParams[myParam][2]
+
+    from numpy import linspace
+    if len(list) is 0:
+        for value in linspace(min,max,entries):
+            tmp = {}
+            tmp[myParam]=value
+            myList.append(tmp)
+    else:
+        for point in list:
+            for value in linspace(min,max,entries):
+                tmp = dict(point)
+                tmp[myParam]=value
+                myList.append(tmp)
+    return myList
+
+def consolidateListType(myList,intTypeList):
+    for i in range (0, len(myList)):
+        for j in range(0,len(intTypeList)):
+            if intTypeList[j] in myList[i]:
+                myList[i][intTypeList[j]] = int(math.floor(myList[i][intTypeList[j]]))
+    return myList
+
+def createJson(pointParam,cfgJson):
+
+    newJson = deepcopy(cfgJson)
+
+    if "layers" in pointParam:
+        newJson["network"]["topology"]["layers"] = pointParam["layers"]
+    if "neurons" in pointParam:
+        newJson["network"]["topology"]["neurons"] = pointParam["neurons"]
+    if "dropout_rate" in pointParam:
+        newJson["network"]["topology"]["dropout"] = pointParam["dropout_rate"]
+    if "epochs" in pointParam:
+        newJson["network"]["epochs"] = pointParam["epochs"]
+    if "batch_size" in pointParam:
+        newJson["network"]["batchSize"] = pointParam["batch_size"]
+    if "learning_rate" in pointParam:
+        newJson["network"]["optimizer"]["lr"] = pointParam["learning_rate"]
+    if "learning_rate_decay" in pointParam:
+        newJson["network"]["optimizer"]["decay"] = pointParam["learning_rate_decay"]
+    if "L2_regularizer" in pointParam:
+        newJson["network"]["topology"]["l2"] = pointParam["L2_regularizer"]
+
+    return newJson
+
+def saveJson(myList,cfgJson,dir):
+    for i in range(0,len(myList)):
```

Comment:
You do not really need to do ```for i in range(0,len(myList))```, change it to ```for pointParam in myList```. It is much more concise and easier to understand. It also means less lines of code since you can get rid of the next line